client/src/infographic/Infographic.js

```diff
@@ -171,15 +129,82 @@ class InfoGraph_ extends React.Component {
   render() {
     const { subject, active_bubble_id } = this.props;
     const {
-      bubble_menu_loading,
-      infographic_loading,
+      loading,
       subject_bubble_defs,
       panel_keys,
       previous_bubble,
       next_bubble,
+      panel_filter_by_table,
+      panel_filter_by_footnotes,
+      total_number_of_panels,
     } = this.state;
 
-    const loading = bubble_menu_loading || infographic_loading;
+    const show_all_panels_bubble_type = !(
+      active_bubble_id === "financial" || active_bubble_id === "people"
+    );
+    const is_panel_filter_empty =
+      _.filter(panel_filter_by_table).length === 0 &&
+      _.filter(panel_filter_by_footnotes).length === 0;
+
+    const { number_of_active_panels, panel_renderers } =
+      !loading &&
+      _.reduce(
```

Comment:
Feels over complicated, doesn't `number_of_active_panels === panel_renderers.length`? Rather than a reduce, couldn't this just be a filter on panel keys (either mapping to the `PanelRenderer` components here or in the render as before).

I also think pretty much all this logic could be abstracted in to the `panel_filter` function/hidden in the `InfographicFilter` component. I.e. default that function to `_.identity` in `Infographic`'s state, any of the filtering logic would come back up from the `set_panel_filter` call back.  `Infographic` would only be responsible for applying the filter function to the panel key list, doesn't need to know anything more about how filtering works.

Reply:
So `number_of_active_panels` is not equal to `panel_renderers.length`, because of static panels. `panel_renderers.length = `number_of_active_panels` + static panels. Static panels like key concepts panel, late resources and late results panels isn't really considered as infographic panel, so I wanted to exclude them from the count of panels.

Reply:
I separated filtering logic into the `InfographicPanelRenderer` (renamed from `InfographicFilter`) to the best of how I interpreted it. Let me know if that looks okay, I'm pretty sure I didn't fully get it, so let me know any parts I didn't quite get

Reply:
Took a stab it on #866, please review that and ask any questions about what I was thinking when I made the changes that I did!